daemons/gptp/common/avbts_message.hpp

```diff
@@ -223,21 +226,50 @@ class PTPMessageCommon {
 
 #pragma pack(push,1)
 
+#define PATH_TRACE_TLV_TYPE 0x8
+
 class PathTraceTLV {
  private:
 	uint16_t tlvType;
-	uint16_t lengthField;
-	ClockIdentity identity;
+	typedef std::list<ClockIdentity> IdentityList;
+	IdentityList identityList;
  public:
-	 PathTraceTLV() {
-		tlvType = PLAT_htons(0x8);
-		lengthField = PLAT_htons(sizeof(identity));
-	}
-	void setClockIdentity(ClockIdentity * id) {
-		identity = *id;
+	PathTraceTLV() {
+		tlvType = PLAT_htons(PATH_TRACE_TLV_TYPE);
+	}
+	void parseClockIdentity(uint8_t *buffer) {
+		int length = PLAT_ntohs(*((uint16_t *)buffer))/PTP_CLOCK_IDENTITY_LENGTH;
+		buffer += sizeof(uint16_t);
+		for(; length > 0; --length) {
+			ClockIdentity add;
+			add.set(buffer);
+			identityList.push_back(add);
```

Comment:
You would only need a 'delete' if there is a 'new'.  The list is not dynamic and cleans itself up when delete is called on the message.